base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/views/ScriptGeneratorViewModel.java

```diff
@@ -447,21 +472,61 @@ public String getText(Object element) {
      * @param btnGetValidityErrors The validity check button to style change.
      * @param btnGenerateScript The generate script button to style change.
      * @param btnSaveParam The save parameters button.
+     * @param btnSaveParamAs The save parameters as button.
      */
-    protected void bindActionProperties(ActionsViewTable viewTable, Button btnGetValidityErrors, Button btnGenerateScript, Button btnSaveParam) {
+    protected void bindActionProperties(ActionsViewTable viewTable, Button btnGetValidityErrors, Button btnGenerateScript, Button btnSaveParam, Button btnSaveParamAs) {
     this.viewTable = viewTable;
     this.btnGetValidityErrors = btnGetValidityErrors;
     this.btnGenerateScript = btnGenerateScript;
     this.btnSaveParam = btnSaveParam;
+    this.btnSaveParamAs = btnSaveParamAs;
     // Remove listeners so as not to bind them twice
     this.scriptGeneratorModel.getScriptGeneratorTable().removePropertyChangeListener(ACTIONS_PROPERTY, actionChangeListener);
     this.scriptGeneratorModel.getScriptGeneratorTable().addPropertyChangeListener(ACTIONS_PROPERTY, actionChangeListener);
+    this.scriptGeneratorModel.getScriptGeneratorTable().addPropertyChangeListener(ACTIONS_PROPERTY, evt -> {
+    	updateParametersFileModifiedStatus();
+	});
     this.scriptGeneratorModel.removePropertyChangeListener(VALIDITY_ERROR_MESSAGE_PROPERTY, actionChangeListener);
     this.scriptGeneratorModel.addPropertyChangeListener(VALIDITY_ERROR_MESSAGE_PROPERTY, actionChangeListener);
     this.scriptGeneratorModel.removePropertyChangeListener(TIME_ESTIMATE_PROPERTY, actionChangeListener);
     this.scriptGeneratorModel.addPropertyChangeListener(TIME_ESTIMATE_PROPERTY, actionChangeListener);
     }
 
+    private void updateParametersFile(String parametersFilePath) {
+	String displayFile = "Current parameters file: " + parametersFilePath;
+	currentParametersFile = parametersFilePath;  	// Update the current parameter file path for Save.
+	unsavedChangesMarker = false;							// Reset unsaved changes
+	firePropertyChange("parametersFile", parametersFileDisplayString, parametersFileDisplayString = displayFile);
+    }
+    
+    private void updateParametersFileModifiedStatus() {
+    	
+    	// Get the action parameters currently in the table
+    	List<Map<JavaActionParameter, String>> tableActionParameters = new ArrayList<Map<JavaActionParameter, String>>();
+    	for (ScriptGeneratorAction action : scriptGeneratorModel.getActions()) {
+    		tableActionParameters.add(action.getActionParameterValueMap());
+    	}
+    	
+    	// Compare them with those of the loaded file
+    	boolean isEqual = tableActionParameters.equals(currentFileActions);
+    	
+    	if (!isEqual) {
+        	if (!unsavedChangesMarker) {
+        		// If table does not match file, and no unsaved changes marker, display it
+        		String displayString = parametersFileDisplayString + " (*)";
```

Comment:
Can we make `" (*)"` a constant in the code to avoid duplication of it in multiple places.